### PR TITLE
[Feature] Dominant Speaker Feature used in calling view update

### PR DESCRIPTION
**Pull request by lpx1989**
## Purpose
Leverage API of dominant speaker feature from calling SDK to update grid view of participants in the call.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

### CODE REVIEW

File: AzureCommunicationUI/Podfile [hunk 1]
```diff
@@ -45,7 +45,7 @@ end
 target 'AzureCommunicationUIDemoApp' do
   project 'AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp.xcodeproj'
   use_frameworks!
-  pod 'AzureCommunicationCalling', '2.2.1'
+  pod 'AzureCommunicationCalling', '2.3.0-beta.4'
```
Comment (Mohtasim) [inline AzureCommunicationUI/Podfile:48]: Is there a reason not to use the 2.3.0 GA build?

**Review by Mohtasim — COMMENTED** [review]
Please fix the pipeline build error

File: AzureCommunicationUI/Podfile (hunk 1 above)
Comment (lpx1989) [reply AzureCommunicationUI/Podfile]: Thx, I resolved it.

Comment (pavelprystinka) [reply AzureCommunicationUI/Podfile]: Converted PR to draft. This feature is waiting for SDK to make a GA release.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift [hunk 2]
```diff
@@ -13,13 +13,16 @@ class CallingSDKEventsHandler: NSObject, CallingSDKEventsHandling {
     var callInfoSubject = PassthroughSubject<CallInfoModel, Never>()
     var isRecordingActiveSubject = PassthroughSubject<Bool, Never>()
     var isTranscriptionActiveSubject = PassthroughSubject<Bool, Never>()
+    var dominantSpeakersSubject: CurrentValueSubject<[String], Never> = .init([])
+    var dominantSpeakersModifiedTimestampSubject: PassthroughSubject<Date, Never> = .init()
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift:17]: It does not look like we need dominantSpeakersModifiedTimestampSubject. We still just use dominantSpeakersModifiedTimestamp = Date() in the reducer.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/RemoteParticipantsEventsAdapter.swift [hunk 3]
```diff
@@ -7,18 +7,16 @@ import AzureCommunicationCalling
 
 class RemoteParticipantsEventsAdapter: NSObject, RemoteParticipantDelegate {
     var onVideoStreamsUpdated: ((AzureCommunicationCalling.RemoteParticipant) -> Void) = {_ in }
-    var onIsSpeakingChanged: ((AzureCommunicationCalling.RemoteParticipant) -> Void) = {_ in }
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/RemoteParticipantsEventsAdapter.swift:10]: Why have we deleted onIsSpeakingChanged?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/Middleware/CallingMiddleware.swift [hunk 4]
```diff
@@ -146,3 +149,10 @@ private func handleAudioSessionAction(_ action: AudioSessionAction,
         break
     }
 }
+
+private func handleRemoteParticipantsAction(_ action: RemoteParticipantsAction,
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/Middleware/CallingMiddleware.swift:153]: Why do we have this empty action?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Builder/ParticipantInfoModelBuilder.swift [hunk 5]
```diff
@@ -32,7 +32,6 @@ struct ParticipantInfoModelBuilder {
                                     isRemoteUser: true,
                                     userIdentifier: participantIdentifier,
                                     status: .idle,
-                                    recentSpeakingStamp: recentSpeakingStamp,
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Builder/ParticipantInfoModelBuilder.swift:35]: recentSpeakingStamp is not not in use

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/Reducer/RemoteParticipantsReducer.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            = remoteParticipantsState.dominantSpeakersModifiedTimestamp
+        switch action {
+        case .remoteParticipantsAction(.dominantSpeakersUpdated(speakers: let newSpeakers)):
+            dominantSpeakers = newSpeakers
+            dominantSpeakersModifiedTimestamp = Date()
+        case .remoteParticipantsAction(.participantListUpdated(participants: let newParticipants)):
+            participantInfoList = newParticipants
+            lastUpdateTimeStamp = Date()
+        case .errorAction(.statusErrorAndCallReset):
+            participantInfoList = []
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/Reducer/RemoteParticipantsReducer.swift:24]: we also need to update lastUpdateTimeStamp = Date() when we set a new participantInfoList = []

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift [hunk 7]
```diff
@@ -54,21 +62,13 @@ class CallingSDKEventsHandler: NSObject, CallingSDKEventsHandling {
...
             let userIdentifier = remoteParticipant.identifier.rawId
-            self.updateRemoteParticipant(userIdentifier: userIdentifier, updateSpeakingStamp: false)
+            self.updateRemoteParticipant(userIdentifier: userIdentifier)
         }
 
         remoteParticipantEventAdapter.onIsMutedChanged = participantUpdate
         remoteParticipantEventAdapter.onVideoStreamsUpdated = participantUpdate
         remoteParticipantEventAdapter.onStateChanged = participantUpdate
-
-        remoteParticipantEventAdapter.onIsSpeakingChanged = { [weak self] remoteParticipant in
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift:64]: We still need to keep remoteParticipantEventAdapter.onIsSpeakingChanged as this is the trigger to update isSpeaking on the remoteParticipant. We only do not need to keep timestamp

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift (hunk 2 above)
Comment (lpx1989) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift]: addressed.thx

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/RemoteParticipantsEventsAdapter.swift (hunk 3 above)
Comment (lpx1989) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/RemoteParticipantsEventsAdapter.swift:10]: added it back now. thx

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Builder/ParticipantInfoModelBuilder.swift (hunk 5 above)
Comment (lpx1989) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Builder/ParticipantInfoModelBuilder.swift:35]: removed it as well.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift (hunk 7 above)
Comment (lpx1989) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift:64]: added it back to the logic.

**Review by ShaunaSong — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Pull request merged**